cmd/cmd_test.go

```diff
@@ -0,0 +1,140 @@
+package cmd
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestNewCommand(t *testing.T) {
+	params := []struct {
+		given           []string
+		expectedCommand Command
+		expectedError   error
+	}{
+		{
+			[]string{"main.go"},
+			Command{},
+			fmt.Errorf("missing argument for origin repository"),
+		},
+		{
+			[]string{"main.go", "organization/origin"},
+			Command{},
+			fmt.Errorf("missing argument for destination repository"),
+		},
+		{
+			[]string{"main.go", "organization/origin", "organization/destination"},
+			Command{"organization/origin", "organization/destination", map[string]string{}},
+			nil,
+		},
+		{
+			[]string{"main.go", "organization/origin", "organization/destination", "-p"},
+			Command{},
+			fmt.Errorf("missing value for flag -p"),
+		},
+		{
+			[]string{"main.go", "organization/origin", "organization/destination", "-p", "prepend"},
+			Command{"organization/origin", "organization/destination", map[string]string{"modify-prepend": "prepend"}},
+			nil,
+		},
+		{
+			[]string{"main.go", "organization/origin", "organization/destination", "--modify-prepend", "prepend"},
+			Command{"organization/origin", "organization/destination", map[string]string{"modify-prepend": "prepend"}},
+			nil,
+		},
+	}
+
+	for _, tt := range params {
+		actualCommand, actualError := NewCommand(tt.given)
+
+		if actualCommand.Origin != tt.expectedCommand.Origin || actualCommand.Destination != tt.expectedCommand.Destination {
+			fmt.Printf("Given: %v\nExpected: %v\nReceived: %v\n", tt.given, tt.expectedCommand, actualCommand)
+			t.Fail()
```

Comment:
We use testify (https://github.com/stretchr/testify) to make these easier to read/write